### PR TITLE
Updates for 4.3.x node release

### PR DESCRIPTION
**Pull request by dg-concordium**
## Purpose

Inform users about changes for 4.3.x node release.

## Changes

Testnet release notes, updates to node topics about catchup file download

## Checklist

- [X] My code follows the style of this project.
- [ ] The code compiles without warnings.
- [ ] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [X] I accept the above linked CLA.

### CODE REVIEW

**Review by concordium-cl — COMMENTED** [review]
Just a general comment: I think it could be good to get some numbers as in the release note for the last node release. Like up to x% faster or y% less memory usage etc. Of course only where it makes sense but it's a good story for marketing to "brag" about.

File: source/mainnet/net/installation/downloads-testnet.rst [hunk 1]
```diff
@@ -147,7 +147,7 @@ Mac
 
 To run a node on macOS, you need a macOS installer package.
 
-   - `Download the Testnet macOS installer package <https://distribution.concordium.software/macos/signed/concordium-node-4.2.3.pkg>`_
+   - `Download the Testnet macOS installer package <https://distribution.concordium.software/macos/signed/concordium-node-4.3.1.pkg>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads-testnet.rst:150]:
```suggestion
   - `Download the Testnet macOS installer package <https://distribution.concordium.software/macos/signed/concordium-node-4.3.1-0.pkg>`_
```

File: source/mainnet/net/resources/release-notes.rst [hunk 2]
```diff
@@ -11,6 +11,34 @@ Sirius Testnet
...
+---------------------
+
+Concordium Node 4.3.1 introduces a number of performance improvements. 
+
+- Account records are no longer constantly retained in memory. Instead, a limited number are retained in a cache. The number of cached accounts defaults to 10000 and can be configured by the ``--accounts-cache-size`` command line argument or the ``CONCORDIUM_NODE_CONSENSUS_ACCOUNTS_CACHE_SIZE`` environment variable.
+- Reduce startup time and memory use further by reducing the amount of block data retained in memory. In particular finalized blocks are no longer stored in memory.
+- Optimize node data structures related to accounts. This reduces node memory use and improves performance.
+- The gRPC API now reports correctly when the sender of a transaction did not have enough funds to cover the transaction costs.
+- Remove obsolete and unused option ``--max-expiry-duration``.
+- Remove transaction logging functionality from the node. It is replaced by an external service. As a consequence the ``transaction-outcome-logging`` family of command line options are removed from the node.
```
Comment (abizjak) [inline source/mainnet/net/resources/release-notes.rst:26]:
```suggestion
- Remove transaction logging functionality from the node. It is replaced by an external `transaction logger <https://github.com/Concordium/concordium-transaction-logger>`_ service. As a consequence the ``transaction-outcome-logging`` family of command line options are removed from the node.
```

Comment (abizjak) [reply source/mainnet/net/resources/release-notes.rst:26]: Please check that the link formatting is correct.

**Review by abizjak — APPROVED** [review]
Looks good.

File: source/mainnet/net/resources/release-notes.rst [hunk 3]
```diff
@@ -11,6 +11,34 @@ Sirius Testnet
 
    Prior to Sirius, the nodes enforced that a transaction could not be deployed until 2 hours before its expiry date. With Sirius, node validation of transactions has been improved and the 2 hour window has been removed.
 
+August 29, 2022
+
+Concordium Node 4.3.1
+---------------------
+
+Concordium Node 4.3.1 introduces a number of performance improvements.
```
Comment (abizjak) [inline source/mainnet/net/resources/release-notes.rst]:
```suggestion
Concordium Node 4.3.1 introduces a number of performance improvements. The effects of these are that the node on mainnet at the current load will use around 1/4 the memory of the `4.2.3` node. Startup can be up to 50% faster, although exact improvements will be platform dependent.
```

Comment (abizjak) [reply source/mainnet/net/resources/release-notes.rst]: Can we add these numbers @dg-concordium ?

**Pull request merged**